### PR TITLE
added widget views variability feature

### PR DESCRIPTION
**Pull request by kisuvorov**

### CODE REVIEW

File: src/features/widgetViews/ui/Radar/widget/views/default/index.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    const legendContainerRef = useRef(null);
+    const copyrightsContainerRef = useRef(null);
+    const {ratio, isInitiated} = useChildRatio({
+        parentRef: rootRef,
+        extraChildrenRefs: [legendContainerRef, copyrightsContainerRef],
+        parentRatio: configuration.width / configuration.height,
+    });
+    return (
+        <Module className={styles.root} ref={rootRef}>
+            <ResponsiveContainer width="100%" aspect={ratio}>
```
Comment (fidi-anton) [inline src/features/widgetViews/ui/Radar/widget/views/default/index.tsx:46]: looks like this is going to be the same for every View, we should deduplicate the code somehow

File: src/shared/lib/function.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import {TimeoutId} from '@/shared/lib/types';
+
+export const debounce = <T extends Function>(cb: T, wait = 10) => {
```
Comment (fidi-anton) [inline src/shared/lib/function.ts:3]: we should use any library instead of reinventing this

**Comment by fidi-anton** [comment]
also, we should add unit tests

File: src/features/widgetViews/ui/LineChart/index.tsx [hunk 3]
```diff
@@ -4,6 +4,7 @@ import Icon from './icon.svg';
 import {getUnavailabilityReason} from './helpers';
 import {WidgetConfig} from '..';
 import {WidgetType} from '../constants';
+import {ViewComponents} from './widget/views'
 
 const lineChart: WidgetConfig = {
```
[the remark this answers is not in the document]
Comment (kisuvorov) [reply src/features/widgetViews/ui/LineChart/index.tsx:9]: This one? https://github.com/fidi-tech/dotsight-ui/pull/36/files#diff-d0f52102656293a3e8612743c69ead25fd7a96c0c2a1100700c4553614f668b9R17

File: src/features/widgetViews/ui/LineChart/widget/views/index.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        timestamp: number,
+    }[];
+    keys: string[];
+    items: Record<SubCategoryId, {id: SubCategoryId, name: string, icon?: string}>;
+    unitId?: UnitId;
+}
+
+export const ViewComponents = {
+    [ViewType.default]: {
+        configuration: defaultConfiguration,
```
[the remark this answers is not in the document]
Comment (kisuvorov) [reply src/features/widgetViews/ui/LineChart/widget/views/index.ts:23]: This configuration is strictly presetted for widget view. But should be available from application. Later we can add something like customization properties, that comes from outer context

File: src/features/widgetViews/ui/Radar/widget/views/default/index.tsx (hunk 1 above)
Comment (kisuvorov) [reply src/features/widgetViews/ui/Radar/widget/views/default/index.tsx:46]: But next view can possibly be constructed differently. For example with another chart library. Right now I see no reason to make stronger abstractions

**Review by fidi-anton — APPROVED** [review]

**Pull request merged**